File: backend/FinancialManager.API/Models/Transaction.cs
using System.Text.Json.Serialization;

namespace FinancialManager.API.Models;

public class Transaction
{
    public int Id { get; set; }

    public string Description { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public TransactionType Type { get; set; }

    public DateTime Date { get; set; }

    public int CategoryId { get; set; }

    [JsonIgnore]
    public Category? Category { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/financial-manager-web/src/types/transaction.ts
export type TransactionType = 'Income' | 'Expense';

export interface Transaction {
  id: number;
  description: string;
  amount: number;
  type: TransactionType;
  date: string;
  categoryId: number;
  categoryName?: string;
  createdAt: string;
}

export interface CreateTransactionDto {
  description: string;
  amount: number;
  type: TransactionType;
  date: string;
  categoryId: number;
}

export interface UpdateTransactionDto {
  description: string;
  amount: number;
  type: TransactionType;
  date: string;
  categoryId: number;
}
